### ISSUE TITLE
initialPollWaitingForConsumerGroupJoin time should be configurable

### ISSUE THREAD

**Issue by authorjapps** [authorjapps/zerocode#462]
From Slack conversation(Click here to see full details):
https://zerocode-workspace.slack.com/archives/C013T2SV151/p1606153188005800

```
...Probably because the issue am running into is during initialPollWaitingForConsumerGroupJoin(consumer) and there the
 consumer poll is hardcoded to 500ms and ofcourse it has retries about 10 times but that does not seem sufficient - may 
be for several reasons that broker or kafka server is slow or due to the latency of network etc - but no matter what - i 
think the consumer client should be configurable to deal with the intialPoll so its resilient enough to poll and wait for the 
kafka response to let the consumer group to join in without timing out

```

Linked issue: #421 , 
Linked PR: https://github.com/authorjapps/zerocode/pull/422

**Comment by dandalavinod** [comment]
changing to 1000ms did the trick .. ran it few times and have not seen the issue; my proposal for the fix is to use the consumercommonconfig setting consumer.pollingTime and use that as a configurable seting for the initial poll for consumer group to join rather than introducing a new configuration param. I believe the default value is 5000ms which should be sufficient and if needed they can increase at common config level and override it by consumerlocalconfig level.

 public static ConsumerRecords initialPollWaitingForConsumerGroupJoin(Consumer consumer, ConsumerCommonConfigs consumerCommonConfigs ) {
        for (int run = 0; run < 20; run++) {
            if (!consumer.assignment().isEmpty()) {
                return new ConsumerRecords(new HashMap());
            }
            ConsumerRecords records = consumer.poll(**Duration.of(consumerCommonConfigs.getPollingTime())**, ChronoUnit.MILLIS));
            if (!records.isEmpty()) {
                return records;
            }
        }

        throw new RuntimeException("\n********* Kafka Consumer unable to join in time *********\n");
    } 

@authorjapps let me know if that seems reasonable and will open a PR to deliver the change.

**Comment by authorjapps** [comment]
>setting consumer.pollingTime and use that as a configurable seting for the initial poll for consumer group to join rather than introducing a new configuration param

yep, agreed. sounds perfect. Please proceed.

**Comment by dandalavinod** [comment]
@authorjapps as i try push and generate a PR: i am getting Can't connect to URI: git@github.com:authorjapps/zerocode - push not permitted. Seems like i do not have push permissions, could you help take a look and get me necessary permissions or let me know if am missing anything. thanks!

**Comment by sparrowV** [comment]
hey @dandalavinod . You cant directly push to original zerocode repo. If everyone could directly push the code, it would be huge mess  🤕  ( to say the least). So what you should do is to fork zerocode original repo ( [how to do forking is described here](https://docs.github.com/en/free-pro-team@latest/github/getting-started-with-github/fork-a-repo) ) , you clone forked repo locally, make changes , push changes to your forked repo and propose pull request from forking repo ( you would see it on forked repo page). Hope this helps

**Comment by authorjapps** [comment]
@dandalavinod , Also you can follow these [quick n easy guidelines](https://github.com/authorjapps/zerocode/blob/master/CONTRIBUTING.md#how-to-submit-a-pull-request-ie-the-change-you-want-to-merge) in [CONTRIBUTING.md](https://github.com/authorjapps/zerocode/blob/master/CONTRIBUTING.md). ✌️

**Comment by dandalavinod** [comment]
delivered changes using PR: https://github.com/authorjapps/zerocode/pull/463

**Comment by dandalavinod** [comment]
@authorjapps how do i add reviewers to my PR? i do not see the ability for me to add reviewers yet - will i not be able to do it until travis CI reports comes back?? https://github.com/authorjapps/zerocode/pull/463

**Comment by authorjapps** [comment]
Available in version 1.3.27 (awaiting release)
ETA: 6-dec-2020, Sunday
